### ISSUE TITLE
462 - Currying 2

### ISSUE THREAD

**Issue by uid11** [type-challenges/type-challenges#467]
```ts
type Curried<NeededArgs extends readonly unknown[], ReturnValue> = <
  Args extends readonly unknown[]
>(
  ...args: Args
) => NeededArgs extends Args
  ? ReturnValue
  : NeededArgs extends [...Args, ...infer Rest]
  ? Curried<Rest, ReturnValue>
  : never;

declare function DynamicParamsCurrying<
  NeededArgs extends readonly unknown[],
  ReturnValue
>(fn: (...args: NeededArgs) => ReturnValue): Curried<NeededArgs, ReturnValue>;
```

**Comment by hubvue** [comment]
```ts
const curried1 = DynamicParamsCurrying((a: string, b: number, c: boolean) => true)
const result = curried1(123, 123, true)  // never
```
@uid11   Although the test case is correct, the above code does not give me error feedback.

As you can surely see, the first parameter of the curried1 function is of string type, which I wrote as number type , but the editor does not indicate an error, even though the result type is never

**Comment by uid11** [comment]
Yes, the original solution does not output errors for the wrong argument types. I've updated it to output errors in such cases.
But this solution does not point to a specific argument that has the wrong type (it only shows that there is an error in some argument).

**Comment by hubvue** [comment]
@uid11 It looks like there's still a lot of room for improvement, so try optional parameters or function overloading to make the type inference more perfect.

**Comment by uid11** [comment]
@hubvue You're right, using overloads allows to specify exact types for the expected arguments.
Although the TS documentation [recommends](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#use-optional-parameters) using optional arguments in such cases, I abandoned them in order not to mix `undefined` into the argument types.
